drivers/openxr/openxr_device.cpp

```diff
@@ -0,0 +1,2386 @@
+	ERR_FAIL_COND_V_MSG(XR_FAILED(result), false, "OpenXR: Failed to enumerate extension properties");
+
+#ifdef DEBUG
+	for (uint32_t i = 0; i < num_supported_extensions; i++) {
+		print_line("OpenXR: Found OpenXR extension ", supported_extensions[i].extensionName);
+	}
+#endif
+
+	return true;
+}
+
+bool OpenXRDevice::is_extension_supported(const char *p_extension) const {
+	for (uint32_t i = 0; i < num_supported_extensions; i++) {
+		if (strcmp(supported_extensions[i].extensionName, p_extension)) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+void OpenXRDevice::copy_string_to_char_buffer(const String p_string, char *p_buffer, int p_buffer_len) {
+	CharString char_string = p_string.utf8();
+	int len = char_string.length();
+	if (len < p_buffer_len - 1) {
+		// was having weird CI issues with strcpy so....
+		memcpy(p_buffer, char_string.get_data(), len);
+		p_buffer[len] = '\0';
+	} else {
+		memcpy(p_buffer, char_string.get_data(), p_buffer_len - 1);
+		p_buffer[p_buffer_len - 1] = '\0';
+	}
+}
+
+bool OpenXRDevice::create_instance() {
+	// Create our OpenXR instance, this will query any registered extension wrappers for extensions we need to enable.
+
+	// Append the extensions requested by the registered extension wrappers.
+	Map<const char *, bool *> requested_extensions;
+	for (OpenXRExtensionWrapper *wrapper : registered_extension_wrappers) {
+		Map<const char *, bool *> wrapper_request_extensions = wrapper->get_request_extensions();
+
+		// requested_extensions.insert(wrapper_request_extensions.begin(), wrapper_request_extensions.end());
+		for (auto &requested_extension : wrapper_request_extensions) {
+			requested_extensions[requested_extension.key] = requested_extension.value;
+		}
+	}
+
+	// Check which extensions are supported
+	enabled_extensions.clear();
+	for (auto &requested_extension : requested_extensions) {
+		if (!is_extension_supported(requested_extension.key)) {
+			if (requested_extension.value == nullptr) {
+				// nullptr means this is a manditory extension so we fail
+				ERR_FAIL_V_MSG(false, "OpenXR: OpenXR Runtime does not support OpenGL extension!");
+			} else {
+				// set this extension as not supported
+				*requested_extension.value = false;
+			}
+		} else if (requested_extension.value != nullptr) {
+			// set this extension as supported
+			*requested_extension.value = true;
+
+			// and record that we want to enable it
+			enabled_extensions.push_back(requested_extension.key);
+		} else {
+			// record that we want to enable this
+			enabled_extensions.push_back(requested_extension.key);
+		}
+	}
+
+	// Get our project name
+	String project_name = GLOBAL_GET("application/config/name");
+
+	// Create our OpenXR instance
+	XrApplicationInfo application_info{
+		"", // applicationName, we'll set this down below
+		1, // applicationVersion, we don't currently have this
+		"Godot Game Engine", // engineName
+		4, // engineVersion
```

Comment:
You should be able to use one of the constants defined in `core/version.h`.

Reply:
Done, combining major, minor and patch into a single int.

Reply:
And before this is asked, no I can't use OpenXRs make version as that results in a 64bit int while this member is a 32bit int.